### DISCUSSION TITLE
nvim_del_user_command must not be called in a lua loop callback

### DISCUSSION THREAD

**Discussion by afonsolage** [mrcjkb/rustaceanvim#331] [Q&A] [answered]
Hello,

I'm getting this error:
```
[ERROR][2024-03-28 18:51:08] ...m/lsp/client.lua:967	"LSP[rust-analyzer]"	"on_error"	{  code = "ON_EXIT_CALLBACK_ERROR",  err = "...vim/lazy/rustaceanvim/lua/rustaceanvim/commands/init.lua:183: E5560: nvim_del_user_command must not be called in a lua loop callback"}
```

Any tips on how to fix this? 

I'm using neovim 0.10

**Comment by mrcjkb** [comment]
Hey :wave: 

It looks like you're using an outdated version of this plugin.
The bug was fixed in version 4.10.2 (https://github.com/mrcjkb/rustaceanvim/pull/267).

**Reply by afonsolage** [reply, marked as answer]
I'm using the latest version, but after checking `:LspLog` I noticed that this error wasn't the root cause:

```
[ERROR][2024-03-28 19:20:36] .../vim/lsp/rpc.lua:789	"rpc"	"rust-analyzer"	"stderr"	"Could not start dynamically linked executable: rust-analyzer\nNixOS cannot run dynamically linked executables intended for generic\nlinux environments out of the box. For more information, see:\nhttps://nix.dev/permalink/stub-ld\n"
[ERROR][2024-03-28 19:20:36] ...m/lsp/client.lua:967	"LSP[rust-analyzer]"	"on_error"	{  code = "ON_EXIT_CALLBACK_ERROR",  err = "...vim/lazy/rustaceanvim/lua/rustaceanvim/commands/init.lua:183: E5560: nvim_del_user_command must not be called in a lua loop callback"}
```
It was a combo of NixOS + Mason, since I was using Mason to install `rust_analyzer` and NixOS was complaining about running some kind of "generic static linked" stuff.

I removed `rust_analyzer` from "ensure_installed" from Mason and everything is fine now.

In the end, it was not caused by rustaceanvim, but as a side effect of the `rust_analyzer` error.

Anyways, thanks for your time!
